Storage.md

```diff
@@ -0,0 +1,230 @@
+
+```typescript
+ItemsHolder.setItem("color", "purple");
+```
+
+To manually update a changed item's value in localStorage, use `saveItem`.
+To save all the items' values in localStorage, use `saveAll`.
+
+```typescript
+ItemsHolder.saveItem("color");
+ItemsHolder.saveAll();
+```
+
+`increase` and `decrease` add and subtract respectively from the item's value.
+`toggle` flips the boolean value of the item.
+
+```typescript
+ItemsHolder.increase("weight", 14);
+ItemsHolder.decrease("weight", 10);
+ItemsHolder.toggle("started");
+```
+
+`checkExistence` can be used to check if there is an item under the specified key and if not, add it to the collection.
+
+```typescript
+ItemsHolder.checkExistence("color");
+```
+
+By default if an item does not exist when `setItem`, `getItem`, `increase`, or `toggle` are called, the item is then added to the collection in the process of the respective function.
+Define `allowNewItems` to explicitly enable or disable this feature.
+
+```typescript
+let ItemsHolder: ItemsHoldr = new ItemsHoldr({ allowNewItems: false });
+```
+
+## Updating
+
+When an item's value changes, the module checks its triggers, modularity, and element status.
+
+Triggers are functions that get run when an item's value equals the specified value.
+
+```typescript
+ItemsHolder.addItem("color", {
+    triggers: {
+        red: () => { this.value = "black"; }
+    }
+});
+ItemsHolder.setItem("color", "red");
+```
+
+Modularity restricts an item's value to a range from 0 to the specified max and runs the `onModular` function as many times until the value hits the modular range.
+
+```typescript
+ItemsHolder.addItem("time", {
+    modularity: 12,
+    num: 0,
+    onModular: () => { this.num += 1; }
+});
+ItemsHolder.setItem("time", 100);
+```
+
+### Auto Save
+
+By default, the values in localStorage are not updated when their values change.
+Auto saving, which updates localStorage values, can be enabled when making the ItemsHolder container.
+
+```typescript
+let ItemsHolder = new ItemsHoldr({ autoSave: true });
+```
+
+To toggle the autoSave feature, use `toggleAutoSave`.
+
+```typescript
+ItemsHolder.toggleAutoSave();
+```
+
+## Clearing and Defaults
+
+To clear all items from the collection, use `clear`.
+
+```typescript
+ItemsHolder.clear();
+```
+
+To clear the collection, but keep some items which will always be in the collection, utilize ItemsHoldr's `valueDefault` property.
+
+```typescript
+let ItemsHolder: ItemsHoldr = new ItemsHoldr({
+    values: {
+        color: {
+            valueDefault: "red"
+        }
+    }
+});
+ItemsHolder.clear(); // "color" is still in the collection with a reset value of "red"
+```
+
+## Elements
+
+### *Note:* This part of ItemsHoldr will be placed into another module in the future.
+
+HTML elements stored within ItemsHoldr will have their values on the page updated to their changed value.
+
+```typescript
+ItemsHolder.addItem("color", {
+    hasElement: true,
+    element: {}
```

Comment:
element has to be defined or else errors about element being undefined happened so I left it at `{}`